### PR TITLE
asset pagination

### PR DESCRIPTION
**Pull request by jsec516**

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -64,5 +64,13 @@ module.exports = {
     }
 
     return this.arrayToCSV(options, ['fields', 'filterValues']);
-  }
+  },
+
+  nextPageType: function(url) {
+    var type = 'token';
+    if(requestedUrl.indexOf('asset') < 0) {
```
Comment (z3dtech) [inline lib/util.js]: isn't this backward? `requestedUrl.indexOf('asset') !== -1`

**Review by z3dtech — CHANGES_REQUESTED** [review]
You will need unit tests for all changes to get approved to merge into package

**Comment by z3dtech** [comment]
Overall looks pretty good

**Comment by z3dtech** [comment]
FYI -- as a temporary solution I rewrote this function in GoLang using a less opinionated package and integrated it into Backstage already https://github.com/FunnelEnvy/backstage-internal-services/pull/30 so you can take your time here and lets get the changes properly merged into the public package before we deploy

File: lib/util.js (hunk 1 above)
Comment (jsec516) [reply lib/util.js]: good catch.. fixed it

**Comment by jsec516** [comment]
@z3dtech  please review the pr  so that we can merge it in master and submit pr to main repo. Test case has been added to ensure that changes are working.

**Review by z3dtech — APPROVED** [review]
approved pending offset testing. When offset is a number other than 100 gotta make sure it doesn't break.

**Pull request merged**